Detailed deletion prompt

Remmbrandt08 opened the pull request:
Specifies exactly what the action being performed does. Modifies BeatmapDeleteDialog.cs to do this. 

Attempts to resolve https://github.com/ppy/osu/issues/30071

stanriders commented on .gitignore:
```diff
@@ -344,3 +344,4 @@ FodyWeavers.xsd
 
 .idea/.idea.osu.Desktop/.idea/misc.xml
 .idea/.idea.osu.Android/.idea/deploymentTargetDropDown.xml
+.vscode/settings.json
```
Please remove

Remmbrandt08 replied:
done

stanriders commented on osu.Game/Screens/Select/BeatmapDeleteDialog.cs:
```diff
@@ -14,7 +14,7 @@ public partial class BeatmapDeleteDialog : DangerousActionDialog
         {
             this.beatmapSet = beatmapSet;
-            BodyText = $@"{beatmapSet.Metadata.Artist} - {beatmapSet.Metadata.Title}";
+            BodyText = $@"You are PERMANENTLY DELETING: {beatmapSet.Metadata.Artist} - {beatmapSet.Metadata.Title}";
```
It's not permanent until you close the game

Remmbrandt08 replied:
that is understandable, but it would be more userfriendly to mention this, regardless of the time of deletion. Could you suggest what you had in mind to replace this with?

stanriders replied:
Just removing the word "permanent" would be fine by me

Remmbrandt08 replied:
done with the latest commit

arorann committed:
detailed deletion prompt

remove stray modification to gitignore

changed dialog text

detailed deletion prompt

remove stray modification to gitignore

changed dialog text

Commit:
Merge branch 'master' into beatmap-deletion

smoogipoo commented on osu.Game/Screens/Select/BeatmapDeleteDialog.cs:
```diff
@@ -14,7 +14,7 @@ public partial class BeatmapDeleteDialog : DangerousActionDialog
         {
             this.beatmapSet = beatmapSet;
-            BodyText = $@"{beatmapSet.Metadata.Artist} - {beatmapSet.Metadata.Title}";
+            BodyText = $@"You are DELETING: {beatmapSet.Metadata.Artist} - {beatmapSet.Metadata.Title}";
```
Needs localisation added.

smoogipoo reviewed (changes requested):
It sounds very aggressive...
- "Are you sure you want to delete <>?"
- "Delete <>?"

Would like second opinion on the appropriate messaging. The dialog with the current header-body style doesn't lend too much freedom here...

There is precedent for the above at least used in the tournament client:
- https://github.com/ppy/osu/blob/5c826be652128b2ae9ae6150abb98f5632ef89aa/osu.Game.Tournament/Screens/Editors/Components/DeleteRoundDialog.cs
- https://github.com/ppy/osu/blob/5c826be652128b2ae9ae6150abb98f5632ef89aa/osu.Game.Tournament/Screens/Editors/Components/DeleteTeamDialog.cs

Maybe copy the messaging there?

smoogipoo commented:
Also, it would be good to create a common class for this:
- https://github.com/ppy/osu/blob/5c826be652128b2ae9ae6150abb98f5632ef89aa/osu.Game/Collections/DeleteCollectionDialog.cs
- https://github.com/ppy/osu/blob/5c826be652128b2ae9ae6150abb98f5632ef89aa/osu.Game/Overlays/Mods/DeleteModPresetDialog.cs
- https://github.com/ppy/osu/blob/5c826be652128b2ae9ae6150abb98f5632ef89aa/osu.Game/Overlays/Settings/Sections/Maintenance/MassDeleteConfirmationDialog.cs
- https://github.com/ppy/osu/blob/5c826be652128b2ae9ae6150abb98f5632ef89aa/osu.Game/Screens/Select/SkinDeleteDialog.cs

bdach commented:
> Also, it would be good to create a common class for this:

These all inherit `DangerousActionDialog` though...? What would you be looking for here?

smoogipoo commented:
My initial idea was something like a `DeleteActionDialog : DangerousActionDialog` if "Delete <>?" were to be used, but it'd be fine to remain as is if there were localisation strings:
- "Delete beatmap <>?"
- "Delete score <>?"

etc... as the tournament client strings are.

bdach commented:
[Time proves to be a flat circle yet again.](https://github.com/ppy/osu/pull/19612)

peppy commented:
Closing this PR
